src/util/worker/mod.rs

```diff
@@ -244,18 +244,24 @@ impl<S: Into<String>> Builder<S> {
 
         Worker {
             scheduler: scheduler,
-            receiver: Mutex::new(Some(rx)),
-            handle: None,
+            handle: Arc::new(Mutex::new(WorkerHandle {
```

Comment:
why change worker?

Reply:
I put some comment before, you can see it above, basically because I need to implement `Clone` for `Worker`. I tried a few ways to deal with clone for `Worker` without changing `Worker` but it didn't work, so I did some modification.

Reply:
Em, seem it is strange to support Clone for a single thread worker...

Reply:
There seems no alternative, because `Storage` implements `Clone`, if I put `Scheduler` in `Storage`, I still have to implement `Clone` for  both `Scheduler` and `Worker` because `Worker` is in `Scheduler`, and it's much more complicated to implement `Clone` for `Scheduler`; if I put `Worker` in `Storage`, I only have to implement `Clone` for `Worker`.